pycqed/instrument_drivers/meta_instrument/qubit_objects/QuDev_transmon.py

```diff
@@ -0,0 +1,210 @@
+
+        self.heterodyne_instr.frequency(previous_freq)
+
+        if analyze:
+            ma.MeasurementAnalysis(auto=True, close_fig=close_fig)
+
+    def measure_homodyne_acqusition_delay(self, delays=None, MC=None,
+                                          analyze=True, close_fig=True):
+        """
+        Varies the delay between the homodyne modulation signal and
+        acquisition. Measures the transmittance.
+        """
+        if delays is None:
+            raise ValueError("Unspecified delays for measure_homodyne_"
+                             "acquisition_delay")
+
+        if MC is None:
+            MC = self.MC
+
+        # set number of averages to 1 due to a readout bug
+        previous_nr_averages = self.heterodyne_instr.nr_averages()
+        self.heterodyne_instr.nr_averages(1)
+        previous_delay = self.heterodyne_instr.acquisition_delay()
+
+        self.prepare_for_continuous_wave()
+        MC.set_sweep_function(pw.wrap_par_to_swf(
+            self.heterodyne_instr.acquisition_delay))
+        MC.set_sweep_points(delays)
+        MC.set_detector_function(det.Heterodyne_probe(
+            self.heterodyne_instr, trigger_separation=5e-6,
+            demod_mode='single'))
+        MC.run(name='acquisition_delay_scan'+self.msmt_suffix)
+
+        self.heterodyne_instr.acquisition_delay(previous_delay)
+        self.heterodyne_instr.nr_averages(previous_nr_averages)
+
+        if analyze:
+            ma.MeasurementAnalysis(auto=True, close_fig=close_fig)
+
+    def measure_spectroscopy(self, freqs=None, MC=None, analyze=True,
+                             close_fig=True, update=False):
+        """ Varies qubit drive frequency and measures the resonator
+        transmittance """
+        if freqs is None:
+            raise ValueError("Unspecified frequencies for "
+                                 "measure_spectroscopy and no previous value")
+
+        if MC is None:
+            MC = self.MC
+
+        self.prepare_for_continuous_wave()
+        self.cw_source.on()
+
+        MC.set_sweep_function(pw.wrap_par_to_swf(self.cw_source.frequency))
+        MC.set_sweep_points(freqs)
+        MC.set_detector_function(det.Heterodyne_probe(
+            self.heterodyne_instr, trigger_separation=2.8e-6,
+            demod_mode='single'))
+        MC.run(name='spectroscopy'+self.msmt_suffix)
+
+        self.cw_source.off()
+
+        if analyze:
+            ma.MeasurementAnalysis(auto=True, close_fig=close_fig)
+
+    def measure_rabi(self):
+        raise NotImplementedError()
+
+    def measure_T1(self):
+        raise NotImplementedError()
+
+    def measure_ramsey(self):
+        raise NotImplementedError()
+
+    def measure_echo(self):
+        raise NotImplementedError()
+
+    def measure_allxy(self):
+        raise NotImplementedError()
+
+    def measure_ssro(self):
+        raise NotImplementedError()
+
+    def find_resonator_frequency(self, update=True, freqs=None, MC=None,
```

Comment:
I would not necessarily define these functions in the init but rather in a script file, but of course that is personal so you should do it however you like :).